### PR TITLE
DAOS-13492 test: further fix mdtest POSIX oclass

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: basic_checkout MdtestSmall
Skip-unit-tests: true
Skip-fault-injection-test: true

When running mdtest with POSIX, oclass needs to be set when creating the container.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mjean308 — APPROVED** [review]

**Review by rpadma2 — DISMISSED** [review]
LGTM.

File: src/tests/ftest/mdtest/small.yaml [hunk 1]
```diff
@@ -35,12 +35,14 @@ pool:
 container:
   type: POSIX
   properties: cksum:crc16,cksum_size:16384,srv_cksum:on
+  oclass: S1
```
Comment (phender) [inline src/tests/ftest/mdtest/small.yaml:38]: file_oclass?

File: src/tests/ftest/util/mdtest_test_base.py [hunk 2]
```diff
@@ -44,19 +44,21 @@ def setUp(self):
...
+            cont_namespace (str, optional): container namespace when creating a container.
+                Defaults to None.
         """
         # Create a pool if one does not already exist
         if self.pool is None:
             self.add_pool(connect=False)
         # create container
         if self.container is None:
-            self.add_container(self.pool)
+            self.container = self.get_container(self.pool, namespace=cont_namespace)
```
Comment (phender) [inline src/tests/ftest/util/mdtest_test_base.py:61]: Maybe this should be a class-specific version of get_conatianer() - get_mdtest_container() - that ensures that the container is created with the oclass params that match the mdtest params.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/mdtest_test_base.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/util/mdtest_test_base.py]: Agree. Updated

**Review by phender — APPROVED** [review]

**Comment by daltonbohning** [comment]
Tests passed with new changes: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12194/2/testReport/

**Review by mjean308 — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by daltonbohning** [comment]
Ran with
`Test-tag: basic_checkout MdtestSmall test_ec_online_rebuild_mdtest test_ec_mdtest_smoke test_aggregation_punching`
And tests passed:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12194/2/testReport/

**Pull request merged**